crates/oxc_linter/src/rules/react/display_name.rs

```diff
@@ -0,0 +1,2402 @@
+            } else if let Expression::StaticMemberExpression(_) = &member.object {
+                // Handle nested case like: Namespace.Component.displayName
+                // We need to extract the full path and compare
+                if let Some(name) = component_name {
+                    let path = extract_member_expression_path(&member.object);
+                    return path == name;
+                }
+            }
+            return true;
+        }
+    }
+
+    false
+}
+
+/// Extract the full path from a member expression (e.g., "Namespace.Component")
+fn extract_member_expression_path(expr: &Expression) -> String {
+    match expr {
+        Expression::Identifier(ident) => ident.name.to_string(),
+        Expression::StaticMemberExpression(member) => {
+            let base = extract_member_expression_path(&member.object);
+            format!("{}.{}", base, member.property.name)
+        }
+        _ => String::new(),
+    }
+}
+
+/// Check if a symbol has a displayName assignment via semantic references
+fn has_display_name_via_semantic(
+    symbol_id: SymbolId,
+    component_name: Option<&CompactStr>,
+    ctx: &LintContext,
+) -> bool {
+    let component_name_str = component_name.map(oxc_span::CompactStr::as_str);
+
+    // Check all references to this symbol
+    for reference in ctx.scoping().get_resolved_references(symbol_id) {
+        if is_display_name_assignment_for_reference(reference, component_name_str, ctx) {
+            return true;
+        }
+    }
+
+    false
+}
+
+/// Check if any write reference to a symbol assigns createContext()
+/// This handles patterns like: var Hello; Hello = createContext();
+fn check_context_assignment_references(
+    symbol_id: SymbolId,
+    decl_node: &AstNode,
+    ctx: &LintContext,
+) -> Option<ReactComponentInfo> {
+    // Only check if the declaration is a VariableDeclarator with no init
+    if !matches!(decl_node.kind(), AstKind::VariableDeclarator(decl) if decl.init.is_none()) {
+        return None;
+    }
+
+    let name = if let AstKind::VariableDeclarator(decl) = decl_node.kind() {
+        decl.id.get_identifier_name().map(|s| CompactStr::from(s.as_str()))
+    } else {
+        None
+    };
+
+    // Check all write references to this symbol
+    for reference in ctx.scoping().get_resolved_references(symbol_id) {
+        if !reference.is_write() {
+            continue;
+        }
+
+        let ref_node = ctx.nodes().get_node(reference.node_id());
+
+        // Walk up to find the assignment expression
+        for ancestor_id in ctx.nodes().ancestor_ids(ref_node.id()) {
+            let ancestor = ctx.nodes().get_node(ancestor_id);
+
+            if let AstKind::AssignmentExpression(assign) = ancestor.kind()
+                && let Expression::CallExpression(call) = &assign.right
+                && let Some(callee_name) = call.callee_name()
+                && (callee_name == "createContext" || callee_name.ends_with(".createContext"))
```

Comment:
it's impossible for `callee_name` to end with `.createContext`

The only case would be if you had code like:

```
foo["bar.createContext"]()
```

but we don't care about that case since you're not accessing the createContext property, you're accessing some other property.

This is the same for the other `ends_with(".createContext")` checks in this file